packages/protocol/contracts/common/Accounts.sol

```diff
@@ -180,6 +195,59 @@ contract Accounts is
     emit AccountMetadataURLSet(msg.sender, metadataURL);
   }
 
+  /**
+   * @notice Set the indexed signer for a specific role
+   * @param signer the address to set as default
+   * @param role the role to register a default signer for
+   */
+  function setDefaultSigner(address signer, string memory role) public {
+    require(
+      isGenericSigner(msg.sender, signer, role),
+      "Must finish authorization before setting default signer"
+    );
+
+    Account storage account = accounts[msg.sender];
+    if (keccak256(abi.encodePacked(role)) == keccak256(abi.encodePacked(VoteSigner))) {
+      account.signers.vote = signer;
```

Comment:
if we are deprecating, why still assign to `signers`?

Reply:
backwards compatibility?

Reply:
Yup

Reply:
though I'm wondering if we should really force everyone to make these comparisons and instead just indicate it legacy roles with an extra parameter or different function?

Reply:
So we are of course making this backwards compatible by continuing to write to `signers`. That means all code that looks at `signers` will continue to work moving forward.

New code that's written can keep doing that if they only ever expect to deal with these roles, however it'd be recommended to check `isSigner` with a given address and role _or_ read the `signerAuthorizations` struct to verify.

Reply:
Great callout that `accounts` is `internal`. The case I was thinking about here external callers looking to verify actions.

How can I as an external service accept a message signed by someone? I could for every role I'm willing to accept call `isSigner`, but a better way would be to get a list of all the signers I'm willing to accept and verify the signature against them. That actually makes me believe the current `{started, completed}` route is also risky, when someone says `getSigners` we should not make them filter for `completed`.

Reply:
While I think it's gonna be a while, I do think we'll eventually be able to migrate everyone over or am I underestimating the cost of doing so?